packages/scenes/src/variables/adhoc/AdHocFiltersVariable.test.tsx

```diff
@@ -107,7 +299,101 @@ describe('AdHocFiltersVariable', () => {
 
       render(<variable.Component model={variable} />);
 
-      expect(variable.state.set.isActive).toBe(true);
+      expect(variable.isActive).toBe(true);
     });
   });
 });
+
+const runRequestMock = {
+  fn: jest.fn(),
+};
+
+let runRequestSet = false;
+
+function setup(overrides?: Partial<AdHocFiltersVariableState>) {
+  const getTagKeysSpy = jest.fn();
+  setDataSourceSrv({
+    get() {
+      return {
+        getTagKeys(options: any) {
+          getTagKeysSpy(options);
+          return [{ text: 'key3' }];
+        },
+        getTagValues() {
+          return [{ text: 'val3' }, { text: 'val4' }];
+        },
+        getRef() {
+          return { uid: 'my-ds-uid' };
+        },
+      };
+    },
+    getInstanceSettings() {
+      return { uid: 'my-ds-uid' };
+    },
+  } as unknown as DataSourceSrv);
+
+  // Workaround because you can only call setRunRequest once
+  runRequestMock.fn = jest.fn();
+
+  if (!runRequestSet) {
+    runRequestSet = true;
+    setRunRequest((ds: DataSourceApi, request: DataQueryRequest): Observable<PanelData> => {
+      runRequestMock.fn(ds, request);
+      return of({ series: [], state: LoadingState.Done, timeRange: getDefaultTimeRange() });
+    });
+  }
+
+  setTemplateSrv(templateSrv);
+
+  const filtersVar = new AdHocFiltersVariable({
+    datasource: { uid: 'my-ds-uid' },
+    name: 'filters',
+    filters: [
+      {
+        key: 'key1',
+        operator: '=',
+        value: 'val1',
+        condition: '',
+      },
+      {
+        key: 'key2',
+        operator: '=',
+        value: 'val2',
+        condition: '',
+      },
+    ],
+    ...overrides,
+  });
+
+  const scene = new EmbeddedScene({
+    $timeRange: new SceneTimeRange(),
+    $variables: new SceneVariableSet({
+      variables: [],
+    }),
+    controls: [filtersVar],
```

Comment:
Should this go into `$variables`?
```suggestion
    $variables: new SceneVariableSet({
      variables: [filtersVar],
    }),
    controls: [],
```

Reply:
well it should be a VariableValueSelectors component